src/auto-evo/steps/ModifyExistingSpecies.cs

```diff
@@ -382,48 +335,37 @@ private void PredatorsOf(List<Species> result, Miche micheTree, Species species)
     }
 
     /// <summary>
-    ///   Returns a new list of all possible species that might emerge in response to the provided pressures,
-    ///   as well as a copy of the original species.
+    ///   Adds a new list of all possible species that might emerge in response to the provided pressures,
+    ///   as well as a copy of the original species to <see cref="mutationsToTry"\>.
     /// </summary>
-    /// <returns>List of viable variants and the provided species</returns>
-    private List<MicrobeSpecies> GenerateMutations(MicrobeSpecies baseSpecies, int amount,
-        List<SelectionPressure> selectionPressures)
+    private void GenerateMutations(MicrobeSpecies baseSpecies, Miche currentMiche, int depth)
     {
-        double totalMP = 100 * worldSettings.AIMutationMultiplier;
+        var inputSpecies = generateMutationsWorkingMemory.GetMutationsAtDepth(depth - 1);
 
-        temporaryMutations1.Clear();
-        temporaryMutations1.Add(Tuple.Create(baseSpecies, totalMP));
-        var viableVariants = temporaryMutations1;
+        pressureStack.Push(currentMiche.Pressure);
 
-        // Auto-evo assumes that the order mutations are applied doesn't matter when it actually can affect
-        // results quite a bit. Maybe they should have weights?
-        tempMutationStrategies.Clear();
+        var outputSpecies = generateMutationsWorkingMemory.GetMutationsAtDepth(depth);
+        outputSpecies.Clear();
+        outputSpecies.AddRange(inputSpecies);
 
-        // Collect and shuffle unique strategies
-        foreach (var selectionPressure in selectionPressures)
-        {
-            foreach (var mutation in selectionPressure.Mutations)
-            {
-                if (!tempMutationStrategies.Contains(mutation))
-                    tempMutationStrategies.Add(mutation);
-            }
-        }
+        mutationSorter.Setup(baseSpecies, pressureStack);
 
+        var mutations = currentMiche.Pressure.Mutations;
         bool lawk = worldSettings.LAWK;
 
-        mutationSorter.Setup(baseSpecies, selectionPressures);
-
-        tempMutationStrategies.Shuffle(random);
+        temporaryMutations1.Clear();
+        temporaryMutations1.AddRange(outputSpecies);
 
-        foreach (var mutationStrategy in tempMutationStrategies)
+        foreach (var mutationStrategy in mutations)
         {
-            var inputSpecies = viableVariants;
+            // temporaryMutations1.Clear();
+            // temporaryMutations1.AddRange(outputSpecies);
```

Comment:
This is currently a draft as there is currently a bug, where Pressures with multiple mutations only have the first one applied. The commented out lines should fix the issue, but when I uncomment them auto-evo breaks in a way I can't seem to track down. It also seems to break in the same way without the clear. I would be grateful if someone could look at it, as I'm stuck.